### PR TITLE
python3Packages.pandas: fix build on 32-bit platforms

### PR DESCRIPTION
**Pull request by misuzu**
## Description of changes

pandas is required to build inscape which is required to build noto-fonts-emoji on armv7l-linux

Upstream issue: https://github.com/pandas-dev/pandas/issues/55789

## Things done

- Built on platform(s)
  - [X] armv7l-linux (native)
  - [X] i686-linux
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by misuzu** [comment]
@ofborg build pkgsi686Linux.python3Packages.pandas

File: pkgs/development/python-modules/pandas/default.nix [hunk 1]
```diff
@@ -227,6 +227,9 @@ buildPythonPackage rec {
     "test_binops"
     # These tests are unreliable on aarch64-darwin. See https://github.com/pandas-dev/pandas/issues/38921.
     "test_rolling"
+  ] ++ lib.optional stdenv.is32bit [
+    # https://github.com/pandas-dev/pandas/issues/55789
```
Comment (pbsds) [inline pkgs/development/python-modules/pandas/default.nix:231]: Could this be updated to `https://github.com/pandas-dev/pandas/issues/37398`? I'm making tooling to scan for closed issues with fixes tagged in a release

Comment (misuzu) [reply pkgs/development/python-modules/pandas/default.nix]: Done!

**Comment by pbsds** [comment]
`pkgsi686Linux.python3Packages.pandas` builds on `x86_64-linux`

File: pkgs/development/python-modules/pandas/default.nix [hunk 2]
```diff
@@ -251,9 +254,6 @@ buildPythonPackage rec {
   ];
 
   meta = with lib; {
-    # https://github.com/pandas-dev/pandas/issues/14866
-    # pandas devs are no longer testing i686 so safer to assume it's broken
-    broken = stdenv.isi686;
```
Comment (pbsds) [inline pkgs/development/python-modules/pandas/default.nix:256]: I think this is nice information to retain. Could you please comment out the `broken`, and rephase the comment to `pandas devs no longer test i686, it's commonly broken`

Comment (misuzu) [reply pkgs/development/python-modules/pandas/default.nix:256]: Done!

**Pull request merged**